src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -954,20 +973,22 @@ internal static bool CanBePassedByReference(BoundExpression expr)
                     return ((BoundIndexerAccess)expr).Indexer.RefKind != RefKind.None;
 
                 case BoundKind.IndexOrRangePatternIndexerAccess:
-                    var patternIndexer = (BoundIndexOrRangePatternIndexerAccess)expr;
-                    var refKind = patternIndexer.PatternSymbol switch
-                    {
-                        PropertySymbol p => p.RefKind,
-                        MethodSymbol m => m.RefKind,
-                        _ => throw ExceptionUtilities.UnexpectedValue(patternIndexer.PatternSymbol)
-                    };
-                    return refKind != RefKind.None;
+                    return CanBePassedByReference(((BoundIndexOrRangePatternIndexerAccess)expr).IndexerAccess);
+
+                case BoundKind.IndexOrRangeIndexerPatternReceiverPlaceholder:
+                case BoundKind.IndexOrRangeIndexerPatternValuePlaceholder:
+                case BoundKind.ListPatternReceiverPlaceholder:
+                case BoundKind.ListPatternUnloweredIndexPlaceholder:
+                case BoundKind.SlicePatternReceiverPlaceholder:
+                case BoundKind.SlicePatternUnloweredRangePlaceholder:
+                    return true;
```

Comment:
>return true; [](http://example.com/codeflow?start=20&length=12)

It is not clear why is this the right value to return? Why do we even run into them?